### PR TITLE
[WIP] Update XCScheme models to conform AutoEquatable

### PR DESCRIPTION
**Pull request by pepibumur**
Resolves https://github.com/xcodeswift/xcproj/issues/200

### Short description 📝
Use Sourcery to make scheme-related models conform the Equatable protocol.

### Solution 📦
- Update models adding the `AutoEquatable` conformance.
- Generate the extensions using Sourcery.

### CODE REVIEW

File: Sources/xcproj/XCScheme.swift [hunk 1]
```diff
@@ -251,7 +251,7 @@ final public class XCScheme {
 
     // MARK: - Build Action
 
-    public class SerialAction {
+    public class SerialAction: AutoEquatable {
```
Comment (pepibumur) [inline Sources/xcproj/XCScheme.swift]: I'm having some issues with Sourcery not generating the extensions for the subclasses of `XCScheme.SerialAction`. Do you know @ilyapuchka if the following `for` loop works for nested types?

```
{% for type in types.implementing.AutoEquatable|class|!annotated:"skipEquality" %}
```

**Comment by pepibumur** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry. <br />
You can find the guidelines at <a href="https://github.com/xcodeswift/contributors/blob/master/CHANGELOG_GUIDELINES.md">CHANGELOG_GUIDELINES.md</a>.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by rudedogg** [comment]
Hey @pepibumur, I'm a noob with the codebase, but could the new automatic equatable/hashable conformance be used instead? https://www.hackingwithswift.com/articles/50/whats-new-in-swift-4-1

**Comment by pepibumur** [comment]
Hey @rudedogg welcome! Unfortunately not because the automatic conformance is for value types, not classes  :/.

**Pull request closed without merging**